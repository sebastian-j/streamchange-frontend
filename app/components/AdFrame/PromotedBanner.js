import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(-180px);
  }
  100% {
    transform: translateY(0px);
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
  top: calc(55vh - ${({ offset }) => offset}px);
`;

const Image = styled.img`
  width: 31vw;
`;

const PromotedBanner = props => {
  const [imgHeight, setImgHeight] = useState(0);
  const isVideo = props.imageUrl.substr(props.imageUrl.length - 3) === 'mp4';

  const onImgLoad = ({ target: img }) => {
    setImgHeight(img.offsetHeight);
  };

  const onVideoLoad = ({ target: video }) => {
    setImgHeight(Math.round(video.videoHeight * 0.6));
  };

  return (
    <PromotedContentWrapper offset={imgHeight}>
      <AdTitle>
        <FormattedMessage {...messages.title} />
      </AdTitle>
      <a href={props.channelUrl} target="_blank">
        {!isVideo && (
          <Image alt={props.title} onLoad={onImgLoad} src={props.imageUrl} />
        )}
        {isVideo && (
          <video
            width="100%"
            autoPlay
            loop
            muted
            onLoadedMetadata={onVideoLoad}
          >
            <source src={props.imageUrl} type="video/mp4" />
          </video>
        )}
      </a>
    </PromotedContentWrapper>
  );
};

PromotedBanner.propTypes = {
  channelUrl: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default PromotedBanner;
