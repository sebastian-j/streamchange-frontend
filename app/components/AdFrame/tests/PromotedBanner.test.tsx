import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import PromotedBanner from '../PromotedBanner';

const shallowRenderer = createRenderer();

describe('<PromotedBanner />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <PromotedBanner
        channelUrl="https://www.youtube.com/channel/UC3GumCi7taJQ0wWbKK-hR2w"
        imageUrl="url"
        testMargins
        title="test-advertisement"
      />,
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
