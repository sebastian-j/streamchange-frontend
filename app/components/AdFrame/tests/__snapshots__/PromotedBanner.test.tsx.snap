// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromotedBanner /> should render and match the snapshot 1`] = `
<styled.div
  margin={1}
  offset={0}
>
  <styled.span
    margin={1}
  >
    <FormattedMessage
      defaultMessage="Advertisement"
      id="streamchange.components.AdFrame.title"
      values={Object {}}
    />
  </styled.span>
  <a
    href="https://www.youtube.com/channel/UC3GumCi7taJQ0wWbKK-hR2w"
    target="_blank"
  >
    <div>
      <styled.img
        alt="test-advertisement"
        onLoad={[Function]}
        src="url"
      />
    </div>
    <styled.div
      margin={1}
    />
    <styled.div
      margin={1}
    />
    <styled.div
      margin={1}
    />
  </a>
</styled.div>
`;
